commands/operator-sdk/cmd/new.go

```diff
@@ -300,11 +312,24 @@ func verifyFlags() error {
 	if operatorType != projutil.OperatorTypeAnsible && generatePlaybook {
 		return fmt.Errorf("value of --generate-playbook can only be used with --type `ansible`")
 	}
+
+	if operatorType != projutil.OperatorTypeHelm {
+		if len(helmChartRef) != 0 {
+			log.Fatal("Value of --helm-chart can only be used with --type `helm`")
+		}
+		if len(helmChartRef) == 0 && len(helmChartRepo) != 0 {
+			log.Fatal("Value of --helm-chart-repo can only be used with --type `helm` and --helm-chart")
+		}
+		if len(helmChartVersion) == 0 && len(helmChartRepo) != 0 {
+			log.Fatal("Value of --helm-chart-version can only be used with --type `helm` and --helm-chart")
+		}
+	}
+
 	if operatorType == projutil.OperatorTypeGo && (len(apiVersion) != 0 || len(kind) != 0) {
 		return fmt.Errorf("operators of type Go do not use --api-version or --kind")
 	}
 
-	if operatorType != projutil.OperatorTypeGo {
+	if operatorType == projutil.OperatorTypeAnsible || operatorType == projutil.OperatorTypeHelm && len(helmChartRef) == 0 {
```

Comment:
Should we add a quick comment here to avoid confusion in the future as to why this is so specific?

Something like this should be good:

> It also makes --api-version and --kind optional if --helm-chart is set, If left unset, --api-version is defaulted to charts.helm.k8s.io/v1alpha1 and --kind is deduced from the fetched chart.